pywbem/cim_operations.py

```diff
@@ -1709,6 +1745,311 @@ def OpenEnumerateInstancePaths(self, ClassName, namespace=None,
                 self.operation_recorder.record_staged()
             return result_tuple
 
+    def IterEnumerateInstances(self, ClassName, namespace=None,
+                               LocalOnly=None,
+                               DeepInheritance=None, IncludeQualifiers=None,
+                               IncludeClassOrigin=None, PropertyList=None,
+                               FilterQueryLanguage=None, FilterQuery=None,
+                               OperationTimeout=None, ContinueOnError=None,
+                               MaxObjectCount=DEFAULT_ITER_MAXOBJECTCOUNT,
+                               **extra):
+        """
+        A generator function to retrieve instances from a WBEM Server.
+        This method frees the user of choices between the multiple
+        EnumerateInstances/OpenEnumerateInstance methods and reduces
+        the enumeration to a pythonic iterator idiom.
+
+        This method performs either the `OpenEnumerationInstances` and
+        `PullInstancesWithPath` operations (pull requests) or the
+        EnumerateInstances operation if the WBEM server does not support the
+        DMTF pull operations. It is an alternate to using `EnumerateInstances`
+        or a `OpenEnumerateInstances`/`PullInstancesWithPath` sequence.
+
+        The `use_pull_operations` attribute of the WBEMConnection constructor
+        may be used to control whether pull or traditional requests are
+        used.
```

Comment:
The two paragraphs above could be improved to talk about the default behavior and that the input argument can be used to override the default. Also, I would use "argument" and not "attribute".
Finally, links could be used where possible.

To address all of that, I suggest to replace the two paragraphs above with these:

```
        This method performs either the
        :meth:`~pywbem.WBEMConnection.OpenEnumerationInstances` and
        :meth:`~pywbem.WBEMConnection.PullInstancesWithPath`
        operations (pull operations) or the
        :meth:`~pywbem.WBEMConnection.EnumerateInstances` operation
        (traditional operation) if the WBEM server does not support the
        pull operations. It is an alternative to using these operations that
        automatically uses the pull operations if supported.

        The `use_pull_operations` argument of the
        :class:`~pywbem.WBEMConnection` constructor can be used to
        override that default behavior to either insist on pull operations or
        to avoid their use.
```